### PR TITLE
R package vignette (tutorial)

### PR DESCRIPTION
**Pull request by ledell**
Issue: https://github.com/h2oai/h2o4gpu/issues/460

### CODE REVIEW

File: src/interface_r/vignettes/getting_started.Rmd [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+# Make predictions
+predictions <- model %>% predict(x)
+
+# Score the model (uses Python scikit-learn score)
+# Not sure yet if we should keep score()
+model %>% score(x, y)
+
+# Compute AUC using Metrics package
+library(Metrics)
```
Comment (terrytangyuan) [inline src/interface_r/vignettes/getting_started.Rmd]: Frasco (my previous colleague) and Ledell! +1 for this but we may want `eval = FALSE` for this piece of code and then manually paste the output here, e.g. Metrics package won't be available during CRAN check which also tries to execute the Rmd files in vignettes.

File: src/interface_r/vignettes/getting_started.Rmd [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+# TO DO: use a different dataset and split into train/test
+
+# Initialize and train the classifier
+model <- h2o4gpu.random_forest_classifier() %>% fit(x, y)
+
+# Make predictions
+predictions <- model %>% predict(x)
+
+# Score the model (uses Python scikit-learn score)
+# Not sure yet if we should keep score()
```
Comment (terrytangyuan) [inline src/interface_r/vignettes/getting_started.Rmd]: h2o4gpu Python package depends on sklearn right? If sklearn is installed along the way I would say we keep this. @navdeep-G What do you think?

File: src/interface_r/vignettes/getting_started.Rmd [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+One thing it includes is the [`score()`](http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html#sklearn.ensemble.RandomForestClassifier.score) method for all the supervised algorithms.
+
+The `score()` method returns different output based on the type of problem.  For example, with a Random Forest classifier, the `score()` function:
+
+- Returns the mean accuracy on the given test data and labels.
+- In multi-label classification, this is the subset accuracy which is a harsh metric since you require for each sample that each label set be correctly predicted.  
+
+TO DO: Finish
+
```
Comment (navdeep-G) [inline src/interface_r/vignettes/getting_started.Rmd]: I think we also need examples per algo in the vignette to get full coverage.

Comment (ledell) [reply src/interface_r/vignettes/getting_started.Rmd]: Good idea.  I will add one for each algo.

**Comment by ledell** [comment]
We should extend the unsupervised section (and show how to use `transform()`) but that can probably wait until the next version.

**Comment by terrytangyuan** [comment]
This looks great in general! I think we should at least showing basic transform() on a couple rows from the dataset for unsupervised models. It’s basically the same as predict() but without options to, e.g. output probabilities. Otherwise the two sections look a little inconsistent.

**Comment by navdeep-G** [comment]
Can you please merge master into this branch?

**Comment by ledell** [comment]
@navdeep-G I am making the fixes that @terrytangyuan suggested about `eval = FALSE` and adding `transform()`.  Will be done in the next 5-10 mins and then we can merge to master.

**Comment by navdeep-G** [comment]
Initial vignette looks good. We can review more in detail today.

**Comment by navdeep-G** [comment]
@ledell No, not merge into master. Merge master into this branch to pick up some fixes.

**Comment by ledell** [comment]
@navdeep-G @terrytangyuan Please have a look.  The code that is commented out is code that does not work due to pending bug fixes (namely [this bug](https://github.com/h2oai/h2o4gpu/issues/496)).

File: src/interface_r/vignettes/getting_started.Rmd [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+```
+
+
+### Truncated Singular Value Decomposition (SVD)
+
+Train a truncated SVD model with 4 components and apply the transformation on a test set.
+
+```{r, eval = FALSE}
+model_tsvd <- h2o4gpu.truncated_svd(n_components = 4) %>% fit(train)
+test_transformed <- model_pca %>% transform(test)
```
Comment (navdeep-G) [inline src/interface_r/vignettes/getting_started.Rmd]: Typo: Should be `model_tsvd`

Comment (ledell) [reply src/interface_r/vignettes/getting_started.Rmd]: Thanks!
